scaladoc-testcases/src/tests/exports1.scala

```diff
@@ -14,18 +14,18 @@ class A: //unexpected
    = 1
   var aVar1: 1
    = 1
-  type HKT[T[_], X] //expected: final type HKT = [T[_], X] =>> HKT[T, X]
+  type HKT[T[_], X] //expected: final type HKT = [T[_], X] =>> a.HKT[T, X]
```

Comment:
I'm not sure I understand why we need the `a`/`A` in theses, could you explain ?

Reply:
This file is part of test consisting of 2 files: `exports1.scala` and `exports2.scala`

The test:
https://github.com/lampepfl/dotty/blob/ef974367932cb148bda152ed056183f319a9a2fc/scaladoc/test/dotty/tools/scaladoc/signatures/TranslatableSignaturesTestCases.scala#L96

`exports2.scala` for reference:
https://github.com/lampepfl/dotty/blob/ef974367932cb148bda152ed056183f319a9a2fc/scaladoc-testcases/src/tests/exports2.scala#L1-L12

So it's testing for exported member signatures in `class B`.

Signatures are rendered based on `TypeRepr` from Tasty.
For example `TypeRepr` for `val x: HKT[List, Int]` rendered in `class B` looks like this:
```scala
AppliedType(
  TypeRef(
    ThisType(TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class tests)),object exports1),class A)), // TypeRepr for class A
    type HKT
  ),
  List(
    TypeRef(TermRef(TermRef(ThisType(TypeRef(NoPrefix,module class <root>)),object scala),package),List), // TypeRepr for List
    TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class <root>)),object scala),Int) // TypeRepr for Int
  )
)
```
Based solely on this representation, signature that should be rendered is `A#HKT[List, Int]`

To be honest, I'm not sure entirely whether this is correct way to render signatures of exported members.
If this is not correct, we may want to apply some additional logic for checking if current signature is of an exported member and render it in a different way, but I'm probably gonna need some help with that.

Reply:
@sjrd or @bishabosha, you are very familiar with tasty, what do you think ?

Reply:
`a.HKT[T, X]` seems absolutely correct to me. Otherwise you would be defining `B.HKT` in terms of itself.

The `A#HKT[List, Int]` seem a bit suspicious to me. `this` prefix substitution (the so-called `asSeenFrom` operation) should be replacing `this.HKT[List, Int]` from `A` by `a.HKT[List, Int]` when created by an `export a._`.

In fact, the `TypeRef` seems even more suspicious. As is, it means `A.this.HKT`, not really `A#HKT`. But that's quite bad if that's the type given to a member of `B`, since such a type is only valid when you're inside `A`. It seems the compiler is doing something wrong here.

From the point of view of Scaladoc, I think it should display `A.this.HKT` for the `TypeRef(ThisType(A), type HKT)`.